lib/spack/docs/index.rst

```diff
@@ -37,21 +37,24 @@ package:
 If you're new to spack and want to start using it, see :doc:`getting_started`,
 or refer to the full manual below.
 
------------------
-Table of Contents
------------------
-
 .. toctree::
    :maxdepth: 2
+   :caption: Documentation and Tutorial
 
    features
    getting_started
    basic_usage
+   workflows
+
+
+.. toctree::
+   :maxdepth: 2
+   :caption: Reference Manual
+
    packaging_guide
    mirrors
    configuration
    developer_guide
-   case_studies
    command_index
    package_list
    API Docs <spack>
```

Comment:
I've been thinking about this. How do you feel about the following structure:
```rst
.. toctree::
   :maxdepth: 2
   :caption: User Documentation

   features
   getting_started
   basic_usage
   workflows
   configuration
   mirrors
   package_list
   command_index

.. toctree::
   :maxdepth: 2
   :caption: Developer Documentation

   contribution_guide  # see note below
   packaging_guide
   developer_guide
   API Docs <spack>
```
@davydden is adding a Contribution Guide in #1664. Right now, it mostly contains git shortcuts, but I plan on adding documentation about how to run all of the tests and how you need the tests to pass in order to get a PR merged.